server/src/main/java/org/opensearch/index/ShardIndexingPressureMemoryManager.java

```diff
@@ -0,0 +1,456 @@
+    private final Logger logger = LogManager.getLogger(getClass());
+
+    /**
+     * Shard operating factor can be evaluated using currentShardBytes/shardLimits. Outcome of this expression is categorized as
+     * lower, optimal and upper boundary, and appropriate action is taken once the below defined threshold values are breached.
+     */
+    public static final Setting<Double> LOWER_OPERATING_FACTOR =
+        Setting.doubleSetting("shard_indexing_pressure.operating_factor.lower", 0.75d, 0.0d,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+    public static final Setting<Double> OPTIMAL_OPERATING_FACTOR =
+        Setting.doubleSetting("shard_indexing_pressure.operating_factor.optimal", 0.85d, 0.0d,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+    public static final Setting<Double> UPPER_OPERATING_FACTOR =
+        Setting.doubleSetting("shard_indexing_pressure.operating_factor.upper", 0.95d, 0.0d,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    /**
+     * This determines the max time elapsed since any request was processed successfully. Appropriate action is taken
+     * once the below below defined threshold value is breached.
+     */
+    public static final Setting<Integer> SUCCESSFUL_REQUEST_ELAPSED_TIMEOUT =
+        Setting.intSetting("shard_indexing_pressure.secondary_parameter.successful_request.elapsed_timeout", 300000,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    /**
+     * This determines the max outstanding request that are yet to be processed successfully. Appropriate
+     * action is taken once the below defined threshold value is breached.
+     */
+    public static final Setting<Integer> MAX_OUTSTANDING_REQUESTS =
+        Setting.intSetting("shard_indexing_pressure.secondary_parameter.successful_request.max_outstanding_requests",
+            100, Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    /**
+     * Degradation for a shard can be evaluated using average throughput of last N requests,
+     * where N being {@link ShardIndexingPressureSettings#REQUEST_SIZE_WINDOW}, divided by lifetime average throughput.
+     * Appropriate action is taken once the outcome of above expression breaches the below defined threshold value is breached.
+     */
+    public static final Setting<Double> THROUGHPUT_DEGRADATION_LIMITS =
+        Setting.doubleSetting("shard_indexing_pressure.secondary_parameter.throughput.degradation_factor", 5.0d, 1.0d,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
+
+    /**
+     * The node level soft limit determines when the secondary parameters for shard is to be evaluated for degradation.
+     */
+    public static final Setting<Double> NODE_SOFT_LIMIT =
+        Setting.doubleSetting("shard_indexing_pressure.primary_parameter.node.soft_limit", 0.7d, 0.0d,
+            Setting.Property.NodeScope, Setting.Property.Dynamic);
```

Comment:
Both these setting names are non-intuitive

Reply:
I have tried grouping the settings in two buckets as ``primary_parameter`` and ``secondary_parameter`` for ease of classification and tuning. Let me know if you have any specific suggestions. Moreover, we will have documentation with description and role for each settings, covering more details, as part of the rollout. This should bring in clarity around usage with examples.